src/main/java/org/candlepin/subscriptions/db/model/TallySnapshot.java

```diff
@@ -281,14 +293,26 @@ public void setUsage(Usage usage) {
     }
 
     @Override
-    public String toString() {
-        return "TallySnapshot{" +
-            "id=" + id +
-            ", snapshotDate=" + snapshotDate +
-            ", granularity=" + granularity +
-            ", accountNumber='" + accountNumber + '\'' +
-            ", productId='" + productId + '\'' +
-            ", serviceLevel='" + serviceLevel + '\'' +
-            ", usage='" + usage + '\'' + '}';
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof TallySnapshot)) {
+            return false;
+        }
+        TallySnapshot that = (TallySnapshot) o;
+        return Objects.equals(snapshotDate, that.snapshotDate) &&
+            Objects.equals(productId, that.productId) &&
+            Objects.equals(ownerId, that.ownerId) &&
+            Objects.equals(accountNumber, that.accountNumber) &&
+            serviceLevel == that.serviceLevel &&
+            usage == that.usage &&
+            granularity == that.granularity;
```

Comment:
I'd recommend replacing this long concatenation with StringBuilder, per https://javapapers.com/java/java-string-vs-stringbuilder-vs-stringbuffer-concatenation-performance-micro-benchmark/.  It may not be so critical for this particular code, but it's a habit we should get into.

Reply:
We're not building a string here, it's conditionals for overriding the `equals` method.

Reply:
Okay, I see now.